### PR TITLE
Report errors with html file template references that don't exist

### PR DESCRIPTION
**Pull request by MichaelRFairhurst**
Plus tests

### CODE REVIEW

File: analyzer_plugin/lib/tasks.dart [hunk 1]
```diff
@@ -28,6 +28,13 @@ class AngularWarningCode extends ErrorCode {
           'CANNOT_PARSE_SELECTOR', 'Cannot parse the given selector');
 
   /**
+   * An error code indicating that the provided selector cannot be parsed.
```
Comment (scheglov) [inline analyzer_plugin/lib/tasks.dart]: The commit should be fixed.

File: analyzer_plugin/lib/src/tasks.dart [hunk 2]
```diff
@@ -153,6 +154,33 @@ final ListResultDescriptor<View> VIEWS_WITH_HTML_TEMPLATES =
         'ANGULAR_VIEWS_WITH_TEMPLATES', View.EMPTY_LIST);
 
 /**
+ * The [Source] referenced by a [View]
+ *
+ * The result is only available for [View]s.
+ */
+final ResultDescriptor<Source> HTML_SOURCE_FOR_VIEW =
```
Comment (scheglov) [inline analyzer_plugin/lib/src/tasks.dart]: This result is not used.

File: analyzer_plugin/lib/src/tasks.dart [hunk 3]
```diff
@@ -794,6 +822,123 @@ class BuildUnitViewsTask extends SourceBasedAnalysisTask
...
 
 /**
+ * A task that computes VIEW_HTML_TEMPLATE_MODIFICATION_TIME based
+ * on a view. This is task must exist because I don't think you
+ * can ask for the input
+ *   VIEWS.of(source).toMapOf(SOURCES).to(MODIFICATION_TIME)
+ * but with this task you can request
+ *   VIEWS.of(source).toMapOf(VIEW_HTML_TEMPLATE_MODIFICATION_TIME)
+ */
+class GetHtmlModificationTimeForView extends AnalysisTask {
```
Comment (scheglov) [inline analyzer_plugin/lib/src/tasks.dart]: I think this is overcomplicated.

I think we could check template source existence in `BuildUnitViewsTask`, see TODO in `_createView`, and so these errors will be included into `VIEWS_ERRORS`.

Comment (MichaelRFairhurst) [inline analyzer_plugin/lib/src/tasks.dart]: Agreed Konstantin, thanks. Didn't realize (based on looking at how missing dart imports are found) that `FileSource` has an `exists()` method. Definitely much much much simpler to report based on that in `BuildUnitViewsTask`.

**Comment by MichaelRFairhurst** [comment]
Closing for Konstantin's recommended approach

**Pull request closed without merging**